docs/Examples/Middleware/Main.purs

```diff
@@ -32,8 +32,8 @@ headerMiddleware :: (HTTPure.Request -> HTTPure.ResponseM) ->
                     HTTPure.Request ->
                     HTTPure.ResponseM
 headerMiddleware router request = do
-  response <- router request
-  HTTPure.response' response.status (header <> response.headers) response.body
+  { status, headers, body } <- router request
+  pure $ { status, headers: header <> headers, body}
```

Comment:
Yeah, I agree. In this case it would have involved using `isString`/`isBinary` and `fromString`/`fromBinary`. I'll see if I can come up with a clean way to write middleware.

Reply:
I think I'll have to add at least `mapHeaders :: (Headers -> Headers) -> Response -> Response`. That's the only clean way I can think of. And if there's `mapHeaders`, maybe also `mapStatus`. `mapBody` would be less useful I guess.

Reply:
eh, yeah why don't we just leave it as is for now and we can revisit down the line before releasing if it feels important?  No need to expand the scope of this PR.